File: sdks/flutter/lib/firstuser_sdk.dart
class FirstUserConfig {
  final String baseUrl;
  final String publicAppId;
  final String backendBaseUrl;

  const FirstUserConfig({
    required this.baseUrl,
    required this.publicAppId,
    required this.backendBaseUrl,
  });
}

class FirstUserSdk {
  FirstUserConfig? _config;

  void init(FirstUserConfig config) {
    _config = config;
  }

  void startPresence(Future<void> Function(String status) sendHeartbeat) {
    sendHeartbeat("live");
  }

  Map<String, dynamic> mountHostedChatWidget(String widgetUrl) {
    return {
      "type": "webview",
      "url": widgetUrl,
    };
  }

  Future<Map<String, dynamic>> startEmbeddedWaitlist(Map<String, dynamic> payload) async {
    if (_config == null) throw Exception("FirstUserSdk not initialized");
    return {
      "endpoint": "${_config!.backendBaseUrl}/api/firstuser/waitlist/start",
      "payload": payload,
    };
  }

  Future<Map<String, dynamic>> setPlanTier(String externalUserId, String planTier) async {
    if (_config == null) throw Exception("FirstUserSdk not initialized");
    return {
      "externalUserId": externalUserId,
      "planTier": planTier,
      "endpoint": "${_config!.backendBaseUrl}/api/firstuser/users/$externalUserId/plan",
    };
  }
}

File: sdks/android-kotlin/src/main/kotlin/com/firstuser/sdk/FirstUserSdk.kt
package com.firstuser.sdk

data class FirstUserConfig(
  val baseUrl: String,
  val publicAppId: String,
  val backendBaseUrl: String,
)

class FirstUserSdk {
  private var config: FirstUserConfig? = null

  fun init(config: FirstUserConfig) {
    this.config = config
  }

  fun startPresence(sendHeartbeat: (String) -> Unit) {
    sendHeartbeat("live")
  }

  fun mountHostedChatWidget(widgetUrl: String): String {
    return widgetUrl
  }

  suspend fun startEmbeddedWaitlist(payload: Map<String, String>): String {
    requireNotNull(config) { "FirstUserSdk not initialized" }
    return "POST /api/firstuser/waitlist/start payload=$payload"
  }

  suspend fun setPlanTier(externalUserId: String, planTier: String): String {
    requireNotNull(config) { "FirstUserSdk not initialized" }
    return "POST /api/firstuser/users/$externalUserId/plan planTier=$planTier"
  }
}
